client/src/sessions/session.rs

```diff
@@ -0,0 +1,9 @@
+use tpm2_rs_base::{errors::TssResult, TpmsAuthCommand, TpmsAuthResponse};
+
+/// Trait for types representing TPM sessions.
+pub trait Session {
+    /// Computes the authorization HMAC for this session.
+    fn get_auth_command(&self) -> TpmsAuthCommand;
+    /// Validates the authorization response for this session.
+    fn validate_auth_response(&self, auth: &TpmsAuthResponse) -> TssResult<()>;
```

Comment:
Note: This will eventually need to either be interior mutability, or `&mut self`, because we'll have to update the last-received nonce from the TPM (and the auth response is how we get that).

Password sessions don't exercise this flow, but the HMAC session type will. So, just an FYI on that for the future.

Reply:
I can see this, but would it be ok to only make it mutable once we start working on hmacs. I didn't create this and maybe when we start adding HMAC/auth we see that we need more than just having mutable reference.

Reply:
Sure, I'm mostly just pointing out for the future. :+1: